src/modm/platform/can/stm32/can.cpp.in

```diff
@@ -152,7 +153,8 @@ modm::platform::Can{{ id }}::initializeWithPrescaler(
 	while ((({{ reg }}->MSR & CAN_MSR_INAK) == CAN_MSR_INAK) and (deadlockPreventer-- > 0))  {
 		// wait for the normal mode
 	}
-	modm_assert(deadlockPreventer > 0, "can", "init", "timeout2", {{ 0 if id == '' else id }});
+	return modm_assert_continue_fail_debug(deadlockPreventer > 0, "can.init2",
+		"CAN::initialize() timed out on leaving initialization!", {{ 0 if id == '' else id }});
```

Comment:
What's the C++ way here for returing error codes? Wrap an integer in `std::optional`?

Reply:
In this case we could just return an int, why should we need `std::optional`?
Or even better: return an `enum class Error {...};`.

Reply:
I'll leave to future Niklas to deal with that. We don't have a good universal error code concept (which would include a way to print the error, requires reflection of sorts?).